tests/components/husqvarna_automower/conftest.py

```diff
@@ -0,0 +1,86 @@
+"""Test helpers for Husqvarna Automower."""
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from aioautomower.model import MowerAttributes, MowerList
+from aioautomower.session import AutomowerSession
+import pytest
+
+from homeassistant.components.husqvarna_automower.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from .common import setup_integration
+
+from tests.common import MockConfigEntry, load_fixture, load_json_value_fixture
+
+TEST_MOWER_ID = "c7233734-b219-4287-a173-08e3643f89f0"
+
+
+@pytest.fixture(scope="session")
+def load_mower_fixture():
+    """Load Fixture data."""
+    return load_json_value_fixture("mower.json", DOMAIN)
+
+
+@pytest.fixture(scope="session")
+def mower_list_fixture(load_mower_fixture):
+    """Generate a mower fixture object."""
+    mower = load_mower_fixture
+    mowers_list = MowerList(**mower)
+    mowers = {}
+    for mower in mowers_list.data:
+        mowers[mower.id] = mower.attributes
+    return mowers
+
+
+@pytest.fixture(scope="session")
+def activity():
+    """Defaults value for activity."""
+    return "PARKED_IN_CS"
+
+
+@pytest.fixture(scope="session")
+def state():
+    """Defaults value for state."""
+    return "RESTRICTED"
+
+
+@pytest.fixture
+async def setup_entity(hass: HomeAssistant, mower_list_fixture, activity, state):
+    """Set up entity and config entry."""
+
+    mower_data: MowerAttributes = mower_list_fixture[TEST_MOWER_ID]
+    if activity:
```

Comment:
Avoid using conditional logic in tests. There should be a single branch. Prefer using `pytest.mark.parametrize` to create different test cases.

A typical refactor here would be to parametrize `activity` so that we can always set it on the mower data.

Reply:
I've removed it.
Parametrized tests are already in use.